### PR TITLE
update dragonbones api doc

### PR DESCRIPTION
**Pull request by zhakesi**

### CODE REVIEW

File: cocos/dragon-bones/ArmatureCache.ts [hunk 1]
```diff
@@ -108,14 +111,22 @@ export class AnimationCache {
 
     constructor () {
     }
-
+    /**
+     * @en Initialization.
+     * @zh 初始化。
+     * @param {ArmatureInfo} armatureInfo @en Armature info @zh 龙骨信息
+     * @param {string} animationName @en animation Name @zh 动画名称
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: - I am not sure if `{ArmatureInfo}` and `{string}` are needed
- Initialization -> Initialization function.
- animation Name -> Animation name. 
- should add `.` or `。` at the end of a sentence

File: cocos/dragon-bones/ArmatureCache.ts [hunk 2]
```diff
@@ -147,15 +161,20 @@ export class AnimationCache {
         this.totalTime = 0;
         this.isCompleted = false;
     }
-
+    /**
+     * @en End of play cached frames.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: I think `Ends playback of cached animation frames` is more suitable as it is a function, should starts with a verb.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 3]
```diff
@@ -147,15 +161,20 @@ export class AnimationCache {
...
     end () {
         if (!this._needToUpdate()) {
             this._armatureInfo!.curAnimationCache = null;
             this.frames.length = this._frameIdx + 1;
             this.isCompleted = true;
         }
     }
-
+    /**
+     * @internal since v3.7.0 this is an engine private function.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: `@internal` -> `@engineInternal`

File: cocos/dragon-bones/ArmatureCache.ts [hunk 4]
```diff
@@ -164,7 +183,11 @@ export class AnimationCache {
             && this.totalTime < MaxCacheTime
             && (toFrameIdx === undefined || this._frameIdx < toFrameIdx);
     }
-
+    /**
+     * @en Update to specified animation frame.
+     * @zh 更新动画到指定帧序列。
+     * @param {number} toFrameIdx @en frame index @zh 帧序列
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: The same as above.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 5]
```diff
@@ -185,32 +208,52 @@ export class AnimationCache {
 
         this.end();
     }
-
+    /**
+     * @en Is inited?
+     * @zh 检查是否已初始化。
+     */
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts:228]: - Should be a whole sentence.
- need `@returns` section

File: cocos/dragon-bones/ArmatureCache.ts [hunk 6]
```diff
@@ -185,32 +208,52 @@ export class AnimationCache {
...
     isInited () {
         return this._inited;
     }
-
+    /**
+     * @en Check if current state is invalid.
+     * @zh 检查当前状态是否为无效。
+     */
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts:237]: - need `@returns` section

File: cocos/dragon-bones/ArmatureCache.ts [hunk 7]
```diff
@@ -185,32 +208,52 @@ export class AnimationCache {
...
     isInvalid () {
         return this._invalid;
     }
-
+    /**
+     * @en Mark all cached frames as invalid.
+     * @zh 将所有缓存帧标记为无效的。
+     */
     invalidAllFrame () {
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts:245]: invalidAllFrame -> invalidateAllFrames. Of course, can rename it in next version.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 8]
```diff
@@ -185,32 +208,52 @@ export class AnimationCache {
...
         this.isCompleted = false;
         this._invalid = true;
     }
-
+    /**
+     * @en Update all cached frames.
+     * @zh 更新所有缓存帧。
+     */
     updateAllFrame () {
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts:253]: updateAllFrame -> updateAllFrames. Of course, can rename it in next version.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 9]
```diff
@@ -185,32 +208,52 @@ export class AnimationCache {
...
             this._enableCacheAttachedInfo = true;
             this.invalidAllFrame();
         }
     }
-
+    /**
+     * @en Update to specified animation frame of armature.
+     * @zh 更新龙骨动画到指定帧序列。
+     * @param {Armature} armature @en armature @zh 指定骨架
+     * @param {number} index @en frame index @zh 帧序列
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: The same as above.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 10]
```diff
@@ -301,7 +344,9 @@ export class AnimationCache {
         this.maxVertexCount = vertexCount > this.maxVertexCount ? vertexCount : this.maxVertexCount;
         this.maxIndexCount = indices.length > this.maxIndexCount ? indices.length : this.maxIndexCount;
     }
-
+    /**
+     * @internal since v3.7.0 this is an engine private function.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: @internal -> engineInternal

File: cocos/dragon-bones/ArmatureCache.ts [hunk 11]
```diff
@@ -419,20 +464,29 @@ export class AnimationCache {
...
+     * @en Enable private cache mode.
+     * @zh 启用私有缓存模式。
+     */
     enablePrivateMode () {
         this._privateMode = true;
     }
 
-    // If cache is private, cache will be destroy when dragonbones node destroy.
+    /**
+     * @en If cache is private, cache will be destroy when dragonbones node destroy.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: If using private cache mode, all cached data will be destroyed when corresponding dragonbone nodes are destroyed.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 12]
```diff
@@ -444,7 +498,9 @@ export class ArmatureCache {
         this._armatureCache = {};
         this._animationPool = {};
     }
-
+    /**
+     * @internal since v3.7.0 this is an engine private function.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: @internal -> @engineInternal

File: cocos/dragon-bones/ArmatureCache.ts [hunk 13]
```diff
@@ -461,15 +517,19 @@ export class ArmatureCache {
         if (armature) armature.dispose();
         delete this._armatureCache[armatureKey];
     }
-
-    // When db assets be destroy, remove armature from db cache.
+    /**
+     * @en When db assets be destroy, remove armature from db cache.
+     * @zh 当db assets销毁时,从cache中移除骨架。
+     */
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts:539]: What's db asset? What's db cache?

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 14]
```diff
@@ -142,12 +142,14 @@ interface BoneIndex extends Number {
  * Armature Display has a reference to a DragonBonesAsset and stores the state for ArmatureDisplay instance,
  * which consists of the current pose's bone SRT, slot colors, and which slot attachments are visible. <br/>
  * Multiple Armature Display can use the same DragonBonesAsset which includes all animations, skins, and attachments. <br/>
+ * Cocos Creator supports DragonBones version to 5.6.300.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts:172]: All versions below 5.6.300 are supported?

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 15]
```diff
@@ -426,13 +428,31 @@ export class ArmatureDisplay extends UIRenderer {
             this._frameCache.enableCacheAttachedInfo();
         }
     }
-
+    /**
+     * @en
+     * The socket nodes contains the attached node be registered Synchronous motion
+     * transformation with bones.
+     * @zh
+     * Socket nodes存储挂载到骨骼上的节点，可以随骨骼做同步运动变换。
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts]: 获取挂在在骨骼上的节点。这些节点xxx

我不清楚具体的路基是什么。但是一个函数应该做一件事情。描述的内容应该是做什么事情。至于节点的描述可以放在单独的一句话，属于补充说明。

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 16]
```diff
@@ -426,13 +428,31 @@ export class ArmatureDisplay extends UIRenderer {
...
+     */
     get socketNodes () { return this._socketNodes; }
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts:474]: And i think should add `readonly` for return type. Of course, can modify in next version.

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 17]
```diff
@@ -553,23 +573,23 @@ export class ArmatureDisplay extends UIRenderer {
         setEnumAttr(this, '_defaultArmatureIndex', this._enumArmatures);
         this._useVertexOpacity = true;
     }
-
+    /**
+     * @en
+     * Gets the instance of CCFactory, CCFactory can create instance of Armature.
+     * @zh
+     * 获取工厂实例，工厂实例可以创建骨架实例。
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts]: It seems this function is not a get like function. It doesn't return any thing.

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 18]
```diff
@@ -786,7 +829,11 @@ export class ArmatureDisplay extends UIRenderer {
         // It has no event object.
         this._eventTarget.emit(EventObject.COMPLETE);
     }
-
+    /**
+     * @en Update animation frame.
+     * @zh 跟新动画序列。
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts]: 跟新 -> 更新

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 19]
```diff
@@ -767,7 +804,11 @@ export class ArmatureDisplay extends UIRenderer {
         this._flushAssembler();
         ArmatureSystem.getInstance().add(this);
     }
-
+    /**
+     * @en Be called when the component state becomes invalid.
+     * instance of ArmatureDisplay will be removed from ArmatureSystem.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts]: instance -Instance

File: cocos/dragon-bones/CCSlot.ts [hunk 20]
```diff
@@ -87,7 +88,10 @@ export class CCSlot extends Slot {
         this._color = new Color();
     }
 
-    // return dragonBones.CCTexture2D
+    /**
+     * @en return dragonBones.CCTexture2D.
```
Comment (minggo) [inline cocos/dragon-bones/CCSlot.ts]: It should be a meaningful sentence.

File: cocos/dragon-bones/CCSlot.ts [hunk 21]
```diff
@@ -96,7 +100,10 @@ export class CCSlot extends Slot {
         }
         return null;
     }
-
+    /**
+     * @en Calculate world matrix of slot.
```
Comment (minggo) [inline cocos/dragon-bones/CCSlot.ts]: Calculate -> Calculates

File: cocos/dragon-bones/CCSlot.ts [hunk 22]
```diff
@@ -107,7 +114,10 @@ export class CCSlot extends Slot {
         }
         this._worldMatrixDirty = false;
     }
-
+    /**
+     * @en Reset slot data and state.
```
Comment (minggo) [inline cocos/dragon-bones/CCSlot.ts]: The same as above.

File: cocos/dragon-bones/CCSlot.ts [hunk 23]
```diff
@@ -107,7 +114,10 @@ export class CCSlot extends Slot {
...
+     * @zh 重置插槽数据和状态
```
Comment (minggo) [inline cocos/dragon-bones/CCSlot.ts]: Miss `。`.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 24]
```diff
@@ -147,15 +173,20 @@ export class AnimationCache {
         this.totalTime = 0;
         this.isCompleted = false;
     }
-
+    /**
+     * @en Ends of play cached frames.
```
Comment (SantyWang) [inline cocos/dragon-bones/ArmatureCache.ts]: I can not tell it's a operation or callback. Did you mean ‘stop playing cached frames’

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 25]
```diff
@@ -58,12 +58,16 @@ enum DefaultCacheMode {
 }
 ccenum(DefaultAnimsEnum);
 
+/**
+ * @en Control animation speed, should be larger than 0.
+ * @zh 控制龙骨动画播放速度，数值应大于0。
```
Comment (SantyWang) [inline cocos/dragon-bones/ArmatureDisplay.ts]: Add a white space between number and chinese.

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 26]
```diff
@@ -58,12 +58,16 @@ enum DefaultCacheMode {
...
+ */
 // eslint-disable-next-line prefer-const,import/no-mutable-exports
 export let timeScale = 1;
 
 /**
  * @en Enum for cache mode type.
- * @zh Dragonbones渲染类型
+ * @zh Dragonbones渲染类型。
```
Comment (SantyWang) [inline cocos/dragon-bones/ArmatureDisplay.ts]: Add a white space between chinese and english.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 27]
```diff
@@ -50,13 +50,17 @@ let _y: number;
 const PER_VERTEX_SIZE = 5;
 // x y z / u v / r g b a
 const EXPORT_VERTEX_SIZE = 9;
-
+/**
+ * @internal Since v3.7.2 this is an engine private interface.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: @internal -> @engineInternal

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 28]
```diff
@@ -426,22 +466,46 @@ export class ArmatureDisplay extends UIRenderer {
...
     protected _defaultArmatureIndexValue: DefaultArmaturesEnum = DefaultArmaturesEnum.default;
+    /**
+     * @en The skeleton data of dragonBones.
+     * @zh DragonBones 的骨骼数据。
+     */
     @serializable
     /* protected */ _dragonAsset: DragonBonesAsset | null = null;
+    /**
+     * @en The skeleton atlas data of dragonBones.
+     * @zh dragonBones 的骨骼纹理数据。
```
Comment (SantyWang) [inline cocos/dragon-bones/ArmatureDisplay.ts]: dragonBones -> DragonBones

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 29]
```diff
@@ -583,19 +646,27 @@ export class ArmatureDisplay extends UIRenderer {
         draw.indexCount = indexCount;
         return draw;
     }
-
+    /**
+     * @en
+     * Destroy render data，be called when need to rebuild render data or component is destroyed.
```
Comment (SantyWang) [inline cocos/dragon-bones/ArmatureDisplay.ts]: will be called

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 30]
```diff
@@ -710,7 +788,7 @@ export class ArmatureDisplay extends UIRenderer {
      * @en
      * The key of dragonbones cache data, which is regard as 'dragonbonesName', when you want to change dragonbones cloth.
      * @zh
-     * 缓存龙骨数据的key值，换装的时会使用到该值，作为dragonbonesName使用
+     * 缓存龙骨数据的key值，换装的时会使用到该值，作为 'dragonbonesName' 使用。
```
Comment (SantyWang) [inline cocos/dragon-bones/ArmatureDisplay.ts]: add a white space between english and chinese.

File: cocos/dragon-bones/CCArmatureDisplay.ts [hunk 31]
```diff
@@ -79,7 +118,10 @@ export class CCArmatureDisplay extends DisplayData implements IEventDispatcher {
         }
         return slot!._armature.display;
     }
-
+    /**
+     * @en Conert pos to parent slot coordination.
```
Comment (SantyWang) [inline cocos/dragon-bones/CCArmatureDisplay.ts]: -> Convert

File: cocos/dragon-bones/CCArmatureDisplay.ts [hunk 32]
```diff
@@ -93,33 +135,52 @@ export class CCArmatureDisplay extends DisplayData implements IEventDispatcher {
         newPos.y = pos.x * worldMatrix.m01 + pos.y * worldMatrix.m05 + worldMatrix.m13;
         return newPos;
     }
-
+    /**
+     * @en Conert pos to world coordination.
```
Comment (SantyWang) [inline cocos/dragon-bones/CCArmatureDisplay.ts]: Convert

File: cocos/dragon-bones/ArmatureCache.ts [hunk 33]
```diff
@@ -108,14 +122,23 @@ export class AnimationCache {
...
+     */
     init (armatureInfo: ArmatureInfo, animationName: string) {
         this._inited = true;
         this._armatureInfo = armatureInfo;
         this._animationName = animationName;
     }
 
-    // Clear texture quote.
+    /**
+     * @en Clear all animation frames cached.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: Clear -> Clears. Please check other codes too.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 34]
```diff
@@ -185,32 +220,54 @@ export class AnimationCache {
...
+     * @returns {Boolean}
+     */
     isInited () {
         return this._inited;
     }
-
+    /**
+     * @en Check if current state is invalid.
+     * @zh 检查当前状态是否为无效。
+     * @returns {Boolean}
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: Should add comments about the meaning of the bool. For example, true means xxx, else false.

File: cocos/dragon-bones/ArmatureCache.ts [hunk 35]
```diff
@@ -185,32 +220,54 @@ export class AnimationCache {
...
             this._enableCacheAttachedInfo = true;
             this.invalidAllFrame();
         }
     }
-
+    /**
+     * @en Update to specified animation frame of armature.
+     * @zh 更新龙骨动画到指定帧序列。
+     * @param {Armature} armature @en Aarmature. @zh 指定骨架。
+     * @param {number} index @en Frame index. @zh 帧序列。
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureCache.ts]: Is `{}` needed for parameters?

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 36]
```diff
@@ -95,13 +99,36 @@ function setEnumAttr (obj, propName, enumDef) {
     CCClass.Attr.setClassAttr(obj, propName, 'enumList', Enum.getList(enumDef));
 }
 
+/**
+ * @en Struct that can store rendering data-related information.
+ * @zh 用于存储渲染数据相关信息的结构体。
+ */
 export interface ArmatureDisplayDrawData {
+    /**
+     * @en Material. @zh 材质。
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts]: A Material instance.

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 37]
```diff
@@ -95,13 +99,36 @@ function setEnumAttr (obj, propName, enumDef) {
...
     indexOffset: number;
+    /**
+     * @en Vertex index count. @zh 顶点索引数量。
+     */
     indexCount: number;
 }
 
+/**
+ * @en DragonBones Socket. Used to attach components to bone nodes and move them together
+ * with bone animations. Users need to specify the bone path that needs to follow the
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts]: Users -> Developers

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 38]
```diff
@@ -553,23 +620,19 @@ export class ArmatureDisplay extends UIRenderer {
...
+    /**
+     * @en Initializes _factory from CCFactory, if golbal factory not exists, will create a new one.
+     * @zh 初始化变量 _factory，如果全局工厂实例不存在将新创建一个工厂实列对象。
+     */
     initFactory () {
         this._factory = CCFactory.getInstance();
     }
-
+    /**
+     * @en Called before the component is loaded.
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts]: Called when the component is loaded or before?

File: cocos/dragon-bones/ArmatureDisplay.ts [hunk 39]
```diff
@@ -786,7 +874,11 @@ export class ArmatureDisplay extends UIRenderer {
         // It has no event object.
         this._eventTarget.emit(EventObject.COMPLETE);
     }
-
+    /**
+     * @en Update animation frame.
+     * @zh 更新动画序列。
+     * @param {number} dt @en Delta time. @zh 时间差。
```
Comment (minggo) [inline cocos/dragon-bones/ArmatureDisplay.ts]: Miss unit.

File: cocos/dragon-bones/AttachUtil.ts [hunk 40]
```diff
@@ -32,35 +32,48 @@ const _tempMat4 = new Mat4();
...
+    private _armature: Armature | null = null;
+    private _armatureNode: Node | null = null;
+    private _armatureDisplay: ArmatureDisplay | null = null;
     constructor () {
 
     }
-
+    /**
+     * @en Initialize parameter settings.
+     * @zh 初始化参数设置。
```
Comment (minggo) [inline cocos/dragon-bones/AttachUtil.ts:55]: Initialize parameter settings. -> Initializes parameters.

File: cocos/dragon-bones/AttachUtil.ts [hunk 41]
```diff
@@ -32,35 +32,48 @@ const _tempMat4 = new Mat4();
...
+     */
     init (armatureDisplay: ArmatureDisplay) {
         this._inited = true;
         this._armature = armatureDisplay._armature;
         this._armatureNode = armatureDisplay.node;
         this._armatureDisplay = armatureDisplay;
     }
-
+    /**
+     * @en Reset parameter settings.
```
Comment (minggo) [inline cocos/dragon-bones/AttachUtil.ts]: Reset parameter settings. -> Resets parameter values.

File: cocos/dragon-bones/CCSlot.ts [hunk 42]
```diff
@@ -87,7 +88,11 @@ export class CCSlot extends Slot {
         this._color = new Color();
     }
 
-    // return dragonBones.CCTexture2D
+    /**
+     * @en Get a Texture2D object from _textureData.
+     * @zh 获取2D纹理对象。
+     * @returns @en If _textureData is null, will return null.  @zh 若 _textureData 为 null，将返回 null。
```
Comment (minggo) [inline cocos/dragon-bones/CCSlot.ts]: If _textureData is null, will return null -> A texture instance if _textureData is not null, otherwize returns null.

File: cocos/dragon-bones/ArmatureDisplay.ts (hunk 14 above)
Comment (zhakesi) [reply cocos/dragon-bones/ArmatureDisplay.ts:172]: > All versions below 5.6.300 are supported?

Depends on the compatibility of the DragonBones runtime library itself.
Most low version resources should be fine.

**Review by minggo — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**